packages/react-magma-dom/src/components/CharacterCounter/CharacterCounter.tsx

```diff
@@ -0,0 +1,139 @@
+import * as React from 'react';
+import styled from '../../theme/styled';
+import { transparentize } from 'polished';
+import { ErrorIcon } from 'react-magma-icons';
+import { FormFieldContainerProps } from '../FormFieldContainer';
+import { ThemeContext } from '../../theme/ThemeContext';
+import { ThemeInterface } from '../../theme/magma';
+import { InputSize } from '../InputBase';
+import { I18nContext } from '../../i18n';
+
+export interface CharacterCounterProps
+  extends Omit<FormFieldContainerProps, 'fieldId'>,
+    React.HTMLAttributes<HTMLDivElement> {
+  hasError?: boolean;
+  inputLength?: number;
+  inputSize?: InputSize;
+  testId?: string;
+  /**
+   * @internal
+   */
+  theme?: ThemeInterface;
+}
+
+function buildFontColor(props: CharacterCounterProps) {
+  if (props.isInverse) {
+    if (props.hasError) {
+      return props.theme.colors.danger200;
+    }
+    return transparentize(0.3, props.theme.colors.neutral100);
+  }
+  if (props.hasError) {
+    return props.theme.colors.danger;
+  }
+}
+
+const StyledCharacterCounter = styled.div<{
```

Comment:
Looks like `StyledCharacterCounter` is the div wrapper around the words `X characters allowed`, not a wrapper for the input. You don't need border-radius or even the flex properties on this. Does the margin between the text and input need to differ depending on the input size?

Reply:
`flex` is still needed for when the glyph appears while characters are over the limit.

Reply:
Should we use the `<InputMessage>` component then? That seems to do the styling and show the icon, then we don't have to handle that here. Could be worth looking into.

- [Component](https://github.com/cengage/react-magma/blob/dev/packages/react-magma-dom/src/components/Input/InputMessage.tsx)
- [Usage example](https://github.com/cengage/react-magma/blob/dev/packages/react-magma-dom/src/components/RadioGroup/index.tsx#L178-L186)

Reply:
Seems like it works just as well, switched `<InputMessage` in place.